consensus/istanbul/backend/backend.go

```diff
@@ -497,9 +498,27 @@ func (sb *Backend) Commit(proposal istanbul.Proposal, aggregatedSeal types.Istan
 		return nil
 	}
 
-	if sb.broadcaster != nil {
-		sb.broadcaster.Enqueue(fetcherID, block)
+	// If no cached result, we import block through fetcher, which requires processing the block one more time.
+	// Else if cache hit, we directly write the block.
+	if result == nil {
+		sb.logger.Warn("state process cache miss, enqueuing the block to fetcher", "number", block.Number(), "hash", block.Hash())
+		if sb.broadcaster != nil {
```

Comment:
I'd prefer to totally remove the `broadcaster` dependency. It's a very different code path for the block thatn directly calling `writeBlockWithState()`. 

So, my recommendation is to call `Verify` here as a way of obtaining the `result`

Reply:
What really matters in `broadcaster.Enqueue()` is process the block and call `writeBlockWithState()`, I think calling `process` and `writeBlockWithState()` will be enough. Calling `Verify` seems a overkill

Reply:
Hey @mcortesi, I'm wondering how does the block get propagated if we are not calling `broadcaster.Enqueue`? As I understand, `broadcaster.Enqueue` eventually results in a call to `ProtocolManager.BroadcastBlock` as does returning a block on the `resultsCh` in the worker, but I don't see how the block gets propagated if we are just calling `writeBlockWithState`